lib/doh.c

```diff
@@ -214,19 +214,23 @@ static void local_print_buf(struct Curl_easy *data,
 /* called from multi.c when this DoH transfer is complete */
 static int doh_done(struct Curl_easy *doh, CURLcode result)
 {
-  struct Curl_easy *data = doh->set.dohfor;
-  struct dohdata *dohp = data->req.doh;
-  /* so one of the DoH request done for the 'data' transfer is now complete! */
-  dohp->pending--;
-  infof(doh, "a DoH request is completed, %u to go", dohp->pending);
-  if(result)
-    infof(doh, "DoH request %s", curl_easy_strerror(result));
+  struct Curl_easy *data;
+
+  data = Curl_multi_get_handle(doh->multi, doh->set.dohfor_id);
+  if(data) {
```

Comment:
If `data` is NULL here, it seems like a good place to add some (debug) logging right? Because it is totally unexpected.

Reply:
Added.